### PR TITLE
Adds cryo habitability warnings & Pressure

### PR DESCRIPTION
**Pull request by WhiteHusky**
_Preview images to be updated later_

image

:cl:
* rscadd: Cryo cells will warn if the internal atmosphere will kill occupants.
* rscadd: Cryo cells show internal atmosphere.

### CODE REVIEW

**Comment by SonixApache** [comment]
direct nerf to AI / stealth antag fuckery. Pump should be enough info.

**Comment by WhiteHusky** [comment]
@SonixApache Probably add something to cause the Cyro to *malfunction* and always show Nominal?

**Comment by WhiteHusky** [comment]
@SonixApache 
image

**Comment by WhiteHusky** [comment]
Emagging yields the same result

**Comment by SonixApache** [comment]
why do you keep misspelling it as cyro

**Comment by WhiteHusky** [comment]
@SonixApache Yeah I'm not sure but it's fixed now.

**Comment by Kurfursten** [comment]
I don't like how the text wraps or even the punctuation. Just "Safe", "Warning", and "Danger" should be enough especially considering it already tells you in red text what the problem is based on pressure or temperature.

**Comment by WhiteHusky** [comment]
@Kurfursten Changed to simple messages.
Also checked if the cell is getting hot. It'd be unlikely but hell someone's gonna do something really stupid and they'll get a confirmation of that stupidity.

**Comment by Riekland** [comment]
but why

Making big warnings won't fix retarded doctors and it will only make people with brains more easily detect antag fuckery

**Comment by WhiteHusky** [comment]
The only reason I added this was during testing on the test map. Apparently the cyro cells were directly connected to the canister with no regulation so using them without them being cold would kill you from the pressure damage. So I thought, _'Oh okay maybe a warning suggesting not putting anyone in this 4000kPa 20C kill box would be a good idea if someone fucked up.'_

I could scrap the entire habitability messages and keep just the pressure considering it'd be nice to know when working with the cell.

**Comment by Pathid** [comment]
Checking the freezer already gives you the pressure in the pipes for what that's worth.

I think, while intuitively it makes sense that the cryo tube would have a pressure gauge, it also makes thematic and gameplay sense that when troubleshooting the cryo it takes some thinking instead of the nature of the problem being directly stated to the player.

**Comment by Irradiation** [comment]
Basically what Pathid said. It's not like fixing an overpressurized cryo is difficult. It's as simple as turning off the pressure pump and turning on the emergency pump (And fixing the temperature). But much like everything in the game, it's a knowledge-check.

 I don't think it's a necessary feature per se because you can already tell SOMETHING is wrong when people inside cryo are actively taking damage. Having the tube itself telling you "hey dude, check that pressure" is rather superfluous and probably won't help experienced players. New players might just go "herp derp pressure is high, what do?" until they realize that the pumps on the floor look really nice.

I daresay that this nerfs stealth sabotage of cryo by basically anybody in the game, not just s*licons.

**Comment by ghost** [comment]
Please no. The freezer already tells you everything you temperature/pressure and it has to be in the same pipenet as the cryo for cryo to function. Don't do anything that the PR does.

This shits all over sabotage for no good reason and no, it doesn't magically get better. The telcomms handholding debug PR shat up telcomms sabotage hard, we don't need more.

**Comment by iFlashYou** [comment]
Where is the balance tag? :^)

File: code/game/machinery/cryo.dm [hunk 1]
```diff
@@ -255,9 +255,29 @@ var/global/list/cryo_health_indicator = list(	"full" = image("icon" = 'icons/obj
 		occupantData["fireLoss"] = occupant.getFireLoss()
 		occupantData["bodyTemperature"] = occupant.bodytemperature
 	data["occupant"] = occupantData;
-
+	data["siliconUser"] = istype(user, /mob/living/silicon) || isAdminGhost(user)
+	data["emagged"] = emagged
 	data["cellTemperature"] = round(air_contents.temperature)
 	data["cellTemperatureStatus"] = "good"
+	if(air_contents.temperature > T0C) // if greater than 273.15 kelvin (0 celcius)
```
Comment (ShiftyRail) [inline code/game/machinery/cryo.dm:262]: I think this can be made a `switch` if you really care to

File: code/game/machinery/cryo.dm [hunk 2]
```diff
@@ -389,6 +438,11 @@ var/global/list/cryo_health_indicator = list(	"full" = image("icon" = 'icons/obj
 	updateUsrDialog()
 	return
 
+/obj/machinery/atmospherics/unary/cryo_cell/proc/Emag(mob/user as mob)
```
Comment (ShiftyRail) [inline code/game/machinery/cryo.dm:441]: Pretty certain there is, already, a generic proc for emagging : 
`emag_act(var/mob/user, var/obj/item/card/emag/E)`

**Comment by DamianX** [comment]
thumbed

**Pull request closed without merging**